linux/LinuxProcessTable.c

```diff
@@ -78,11 +78,68 @@ static FILE* fopenat(openat_arg_t openatArg, const char* pathname, const char* m
    return fp;
 }
 
+static long fast_strtol_dec(char** str, int maxlen) {
+   long result = 0;
+   bool neg = false;
+
+   if (**str == '-') {
+      neg = true;
+      (*str)++;
+   }
+
+   if (!maxlen)
+      maxlen = 20; // −9223372036854775808
+
+   while (maxlen-- && **str >= '0' && **str <= '9') {
+      result *= 10;
+      result += **str - '0';
+      (*str)++;
+   }
+
+   return neg ? -result : result;
+}
+
+static unsigned long fast_strtoul_dec(char** str, int maxlen) {
+   unsigned long result = 0;
+
+   if (!maxlen)
+      maxlen = 20; // 18446744073709551615
+
+   while (maxlen-- && **str >= '0' && **str <= '9') {
+      result *= 10;
+      result += **str - '0';
+      (*str)++;
+   }
+
+   return result;
+}
+
+static long long fast_strtoll_dec(char** str, int maxlen) {
+   long long result = 0;
+   bool neg = false;
+
+   if (**str == '-') {
+      neg = true;
+      (*str)++;
+   }
+
+   if (!maxlen)
+      maxlen = 20; // −9223372036854775808
+
+   while (maxlen-- && **str >= '0' && **str <= '9') {
+      result *= 10;
+      result += **str - '0';
+      (*str)++;
+   }
```

Comment:
This "while" loop can be converted to a `fast_strtoull_dec(str, maxlen)` and then cast the result from unsigned to signed.